diff --git a/include/dav1d/dav1d.h b/include/dav1d/dav1d.h
index e8f0705..ff48ea0 100644
--- a/include/dav1d/dav1d.h
+++ b/include/dav1d/dav1d.h
@@ -34,7 +34,7 @@
 #include "common.h"
 #include "picture.h"
 #include "data.h"
-#include "version.h"
+#include "dav1d_version.h"
 
 #ifdef __cplusplus
 extern "C" {
diff --git a/include/dav1d/meson.build b/include/dav1d/meson.build
index 68faaf9..09b0239 100644
--- a/include/dav1d/meson.build
+++ b/include/dav1d/meson.build
@@ -28,7 +28,7 @@ version_h_data.set('DAV1D_API_VERSION_MAJOR', dav1d_api_version_major)
 version_h_data.set('DAV1D_API_VERSION_MINOR', dav1d_api_version_minor)
 version_h_data.set('DAV1D_API_VERSION_PATCH', dav1d_api_version_revision)
 version_h_target = configure_file(input: 'version.h.in',
-                                  output: 'version.h',
+                                  output: 'dav1d_version.h',
                                   configuration: version_h_data)
 
 dav1d_api_headers = [
diff --git a/src/meson.build b/src/meson.build
index 3a34e76..53ef660 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -335,7 +335,7 @@ libdav1d = library('dav1d',
 )
 
 dav1d_dep = declare_dependency(link_with: libdav1d,
-    include_directories : include_directories('../include/dav1d')
+    include_directories : include_directories('../include/', '../include/dav1d/')
 )
 
 #
