diff --git a/meson.build b/meson.build
index 6017b15..eec9249 100644
--- a/meson.build
+++ b/meson.build
@@ -2,10 +2,7 @@ project('BestSource', 'cpp',
     default_options: ['buildtype=release', 'b_ndebug=if-release', 'cpp_std=c++17'],
     license: 'MIT',
     meson_version: '>=0.53.0',
-    version: '.'.join([
-        run_command('grep', 'BEST_SOURCE_VERSION_MAJOR', 'src/version.h', check: true).stdout().strip().split()[2],
-        run_command('grep', 'BEST_SOURCE_VERSION_MINOR', 'src/version.h', check: true).stdout().strip().split()[2],
-    ])
+    version: '6.0'
 )
 
 api_sources = files(
