diff --git a/meson.build b/meson.build
index f7bdbda..3351e53 100644
--- a/meson.build
+++ b/meson.build
@@ -1,21 +1,15 @@
 project('BestSource', 'cpp',
-    default_options: ['buildtype=release', 'b_lto=true', 'b_ndebug=if-release', 'cpp_std=c++17'],
+    default_options: ['buildtype=release', 'b_ndebug=if-release', 'cpp_std=c++17'],
     license: 'MIT',
     meson_version: '>=0.53.0',
-    version: '.'.join([
-        run_command('grep', 'BEST_SOURCE_VERSION_MAJOR', 'src/version.h', check: true).stdout().strip().split()[2],
-        run_command('grep', 'BEST_SOURCE_VERSION_MINOR', 'src/version.h', check: true).stdout().strip().split()[2]
-    ])
 )
 
 link_static = get_option('link_static')
 
 sources = [
     'src/audiosource.cpp',
-    'src/avisynth.cpp',
     'src/bsshared.cpp',
     'src/tracklist.cpp',
-    'src/vapoursynth.cpp',
     'src/videosource.cpp'
 ]
 
@@ -46,10 +40,7 @@ if host_machine.cpu_family().startswith('x86')
     )
 endif
 
-vapoursynth_dep = dependency('vapoursynth', version: '>=55').partial_dependency(compile_args: true, includes: true)
-
 deps = [
-    vapoursynth_dep,
     dependency('libavcodec', version: '>=60.31.0', static: link_static),
     dependency('libavformat', version: '>=60.16.0', static: link_static),
     dependency('libavutil', version: '>=58.29.0', static: link_static),
@@ -65,12 +56,12 @@ elif is_gnu_linker
     link_args += ['-Wl,-Bsymbolic']
 endif
 
-shared_module('bestsource', sources,
+bs_lib = static_library('bestsource', sources,
     cpp_args: ['-D_FILE_OFFSET_BITS=64'],
     dependencies: deps,
     gnu_symbol_visibility: 'hidden',
-    install: true,
-    install_dir: vapoursynth_dep.get_variable(pkgconfig: 'libdir') / 'vapoursynth',
     link_args: link_args,
     link_with: libs
 )
+
+bestsource_dep = declare_dependency(link_with: bs_lib, include_directories: include_directories('src'))
